src/iface-opc-ua.c

```diff
@@ -0,0 +1,347 @@
+/**
+ * This file is part of the KNOT Project
+ *
+ * Copyright (c) 2020, CESAR. All rights reserved.
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ */
+
+
+#include <errno.h>
+#include <stdbool.h>
+#include <string.h>
+#include <knot/knot_protocol.h>
+#include <ell/ell.h>
+#include <knot/knot_types.h>
+
+#include <open62541/client.h>
+#include <open62541/client_config_default.h>
+#include <open62541/client_highlevel.h>
+#include <open62541/client_subscriptions.h>
+#include <open62541/plugin/log_stdout.h>
+
+#include "conf-device.h"
+#include "iface-opc-ua.h"
+
+#define RECONNECT_TIMEOUT 5
+
+#define STRING_EMPTY 0
```

Comment:
What do you think of STRING_EMPTY_LEN? So it will be clearer that this macro contains the length of an empty string.